### PR TITLE
feat: use `UploadFile` for file service uploaded files persistence

### PR DESCRIPTION
**Pull request by foxsin10**

### CODE REVIEW

File: Maskbook/Arweave/ArweaveUploader.swift [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+//  Maskbook
+//
+//  Created by xiaojian sun on 2022/6/16.
+//  Copyright © 2022 dimension. All rights reserved.
+//
+
+import Foundation
+import CryptoSwift
+
+class ArweaveUploader {
```
Comment (foxsin10) [inline Maskbook/Arweave/ArweaveUploader.swift]: Why we need this class?

Comment (sxiaojian88) [reply Maskbook/Arweave/ArweaveUploader.swift:12]: In `ArweaveUploadHandler`  I faced some restrictions.
I have to pass muti parameter in some functions, such as option , item , tx.
And It's a little diffcult to implement progress.
In `ArweaveUploader` I have full freedom to do things.

Comment (sxiaojian88) [reply Maskbook/Arweave/ArweaveUploader.swift:12]: The protocol `FileServiceUploadHandler` function without mutating give these restrictions.
After I finish `ArweaveUploader `, I may move the code to `ArweaveUploadHandler `

Comment (foxsin10) [reply Maskbook/Arweave/ArweaveUploader.swift]: As we are in the context of `Task.detached { }` now, you can have all you need, `mutating` is not necessary. And to make sure the state objects in upload task `Sendable`, it is much better avoid using `Class` implementation.

File: Maskbook/Scene/Personas/Personas/PersonasViewController.swift [hunk 2]
```diff
@@ -201,6 +199,21 @@ class PersonasViewController: BaseViewController {
             }
             .store(in: &disposeBag)
     }
+    
+    func updateSegmentTopConstraint(isSearching: Bool) {
+        let constant: CGFloat = {
```
Comment (foxsin10) [inline Maskbook/Scene/Personas/Personas/PersonasViewController.swift:204]: Maybe this is more readable?
```swift
        let constant: CGFloat = {
            let searchingMargin: CGFloat = {
                if #available(iOS 15, *) {
                    return -100
                } else {
                    return -self.segmentViewController.segmentHeight
                }
            }()
            return isSearching
            ? searchingMargin
            : 146
        }()
```

**Review by sxiaojian88 — APPROVED** [review]

**Pull request merged**